src/helpers/createTimeoutController.ts

```diff
@@ -1,8 +1,27 @@
-import { WaitOptions } from '../types'
+import { jestFakeTimersAreEnabled } from './jestFakeTimersAreEnabled'
 
-function createTimeoutController(timeout: WaitOptions['timeout']) {
+const DEFAULT_TIMEOUT = 1000
+
+function createTimeoutController(timeout: number | false, options: { allowFakeTimers: boolean }) {
   let timeoutId: NodeJS.Timeout
   const timeoutCallbacks: Array<() => void> = []
+  let finished = false
+
+  const { allowFakeTimers = false } = options
+
+  const advanceTime = async (currentMs: number) => {
+    // eslint-disable-next-line no-negated-condition
+    if (currentMs < (!timeout ? DEFAULT_TIMEOUT : timeout)) {
```

Comment:
when timeout is false we have to give it a DEFAULT_TIMEOUT
for test 'should not reject when waiting for next update if timeout has been disabled'